File: module-server/src/main/java/de/twomartens/wahlrecht/configuration/InterceptorConfiguration.java
package de.twomartens.wahlrecht.configuration;

import de.twomartens.wahlrecht.interceptor.HeaderInterceptorRest;
import de.twomartens.wahlrecht.interceptor.LoggingInterceptorRest;
import java.time.Clock;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class InterceptorConfiguration {

  @Bean
  public LoggingInterceptorRest loggingInterceptorRest(Clock clock) {
    return new LoggingInterceptorRest(clock);
  }

  @Bean
  public HeaderInterceptorRest headerInterceptorRest() {
    return new HeaderInterceptorRest();
  }

}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/configuration/InterceptorConfiguration.kt
package de.twomartens.wahlrecht.configuration

import de.twomartens.wahlrecht.interceptor.HeaderInterceptorRest
import de.twomartens.wahlrecht.interceptor.LoggingInterceptorRest
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.time.Clock

@Configuration
open class InterceptorConfiguration {
    @Bean
    open fun loggingInterceptorRest(clock: Clock): LoggingInterceptorRest {
        return LoggingInterceptorRest(clock)
    }

    @Bean
    open fun headerInterceptorRest(): HeaderInterceptorRest {
        return HeaderInterceptorRest()
    }
}